src/Business Foundation/App/NoSeries/src/Batch/NoSeriesBatch.Codeunit.al

```diff
@@ -0,0 +1,206 @@
+    procedure GetLastNoUsed(NoSeriesCode: Code[20]): Code[20]
+    begin
+        exit(NoSeriesBatchImpl.GetLastNoUsed(NoSeriesCode));
+    end;
+
+    /// <summary>
+    /// Get the last number used in the No. Series.
+    /// </summary>
+    /// <param name="NoSeriesLine">The No. Series line to use.</param>
+    /// <returns>The last number used in the series.</returns>
+    procedure GetLastNoUsed(var NoSeriesLine: Record "No. Series Line"): Code[20]
+    begin
+        exit(NoSeriesBatchImpl.GetLastNoUsed(NoSeriesLine));
+    end;
+
+    /// <summary>
+    /// Verifies that the No. Series allows using manual numbers.
+    /// Note: This function allows manual numbers for blank No. Series Codes.
+    /// </summary>
+    /// <param name="NoSeriesCode">Code for the No. Series.</param>
+    procedure TestManual(NoSeriesCode: Code[20])
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        NoSeries.TestManual(NoSeriesCode);
+    end;
+
+    /// <summary>
+    /// Verifies that the No. Series allows using manual numbers and throws an error for the document no. if it does not.
+    /// Note: This function allows manual numbers for blank No. Series Codes.
+    /// </summary>
+    /// <param name="NoSeriesCode">Code for the No. Series.</param>
+    /// <param name="DocumentNo">Document No. to be shown in the error message.</param>
+    procedure TestManual(NoSeriesCode: Code[20]; DocumentNo: Code[20])
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        NoSeries.TestManual(NoSeriesCode, DocumentNo);
+    end;
+
+    /// <summary>
+    /// Simulate the specified No. Series at the specified date starting with the indicated number.
+    /// </summary>
+    /// <param name="NoSeriesCode">Code for the No. Series.</param>
+    /// <param name="UsageDate">The date of retrieval, this will influence which line is used.</param>
+    /// <param name="LastNoUsed">Simulate this is the last number used.</param>
+    /// <returns></returns>
+    procedure SimulateGetNextNo(NoSeriesCode: Code[20]; UsageDate: Date; LastNoUsed: Code[20]): Code[20]
+    var
+        NoSeriesBatchImplSim: Codeunit "No. Series - Batch Impl.";
+    begin
+        exit(NoSeriesBatchImplSim.SimulateGetNextNo(NoSeriesCode, UsageDate, LastNoUsed));
+    end;
+
+    /// <summary>
+    /// Puts the codeunit in simulation mode which disables the ability to save state.
+    /// </summary>
+    procedure SetSimulationMode()
+    begin
+        NoSeriesBatchImpl.SetSimulationMode();
+    end;
+
+    /// <summary>
+    /// Save the state of the No. Series Line to the database.
+    /// </summary>
+    /// <param name="TempNoSeriesLine">No. Series Line we want to save state for.</param>
+    procedure SaveState(TempNoSeriesLine: Record "No. Series Line" temporary);
+    begin
+        NoSeriesBatchImpl.SaveState(TempNoSeriesLine);
+    end;
+
+    /// <summary>
+    /// Save all changes to the database.
+    /// </summary>
+    procedure SaveState();
```

Comment:
Don't we also need methods like `RetrieveState` and `ResetState`?

Reply:
The codeunit automatically retrieves any lines needed and you can reset the state with Clear(object).

Reply:
We automatically retrieve state and to reset state, just use Clear(Object).